dali/base/dadfs.cpp

```diff
@@ -3690,10 +3690,37 @@ protected: friend class CDistributedFilePart;
     {
         clusters.clear();
         unsigned nc = fdesc->numClusters();
-        if (nc) {
-            for (unsigned i=0;i<nc;i++) {
-                IClusterInfo &cluster = OLINK(*fdesc->queryClusterNum(i));
-                clusters.append(cluster);
+        if (nc)
+        {
+            for (unsigned i=0;i<nc;i++)
+            {
+                StringBuffer cname;
+                fdesc->getClusterGroupName(i, cname, nullptr);
+                IClusterInfo *cluster;
+                if (cname.length())
+                    cluster = LINK(fdesc->queryClusterNum(i));
+                else
+                {
+                    // NB: this is non-standard, for situations where the cluster name is not known,
+                    // which happens where none has been provided/set to the file descriptor,
+                    // and the file descriptor has been built up of parts with ips. 
+                    // createClusterInfo will perform a reverse lookup to Dali to try to discover
+                    // a group name.
+                    cluster = createClusterInfo(
+                                  nullptr,
+                                  fdesc->queryClusterGroup(i),
+                                  fdesc->queryPartDiskMapping(i),
+                                  &queryNamedGroupStore()
+                               );
+                }
+#ifdef EXTRA_LOGGING
+                PROGLOG("setClusters(%d,%s)", i, cluster->queryGroupName());
+#endif
+
+                if (!cluster->queryGroup(&queryNamedGroupStore())) {
```

Comment:
Is this going to cause a connection to dali, which you were trying to avoid?  Should it be inside the else.

Reply:
I think it's probably harmless, there's always a group by this stage, so it will never try to resolve,
but it logically belongs more in the else